src/controllers/wallet/wallet.controller.js

```diff
@@ -319,37 +318,42 @@ async function decodeTx(req, res) {
       tx = partial.getTx();
     }
 
+    const getToken = (utxo, txObj) => {
+      if (utxo.token) return utxo.token;
+      if (utxo.token_data === 0) return HATHOR_TOKEN_CONFIG.uid;
+
+      const tokenIndex = (utxo.token_data & TOKEN_INDEX_MASK) - 1;
+      if (txObj.tokens.length > tokenIndex) return txObj.tokens[tokenIndex];
+      return undefined;
+    };
+
     const data = {
+      version: tx.version,
+      type: tx.getType(),
       tokens: tx.tokens,
-      inputs: await tx.inputs.reduce(async (acc, input) => {
-        // the accumulator must be awaited to enforce the sequence processing
-        const results = await acc;
-
+      inputs: await Promise.all(tx.inputs.map(async input => {
```

Comment:
This `Promise.all` can cause some trouble in the future if the number of inputs is big enough. It would add many async tasks that uses blocking resources to the event loop, and Node may not handle it well. This risk is tied to the `getTx` that may query the fullnode all at once. Can we use a `for await (const of tx.inputs)`? `tx.inputs` should be a an iterable.

Reply:
Thats a good point, changed to a for loop that awaits each response before going to the next.